src/robot/libraries/String.py

```diff
@@ -229,13 +256,19 @@ def format_string(self, template, /, *positional, **named):
                 template = reader.read()
         return template.format(*positional, **named)
 
-    def get_line_count(self, string):
+    def get_line_count(self, string: "str | bytes") -> int:
         """Returns and logs the number of lines in the given string."""
         count = len(string.splitlines())
         logger.info(f"{count} lines.")
         return count
 
-    def split_to_lines(self, string, start=0, end=None):
+    # does not work with bytes: ${{b'hello\nworld'}} -> Length of '["b'hello\\nworld'"]' should be 2 but is 1.
+    def split_to_lines(
+        self,
+        string: str,
+        start: "int | str | None" = 0,
+        end: "int | str | None" = None,
+    ) -> list[str]:
```

Comment:
This needs to be `"list[str | bytes]"`. Quotes are needed for Python 3.8 compatibility and `bytes` is needed because this keyword ought to work also with bytes and in that case the result is a list of bytes.

Check also other possible cases where you've used parameterized types like `list[str]` and make sure they are quoted.